### PR TITLE
feat: implement lazy loading for videojs with dynamic import

### LINKED ISSUE #648: only load Video and Image Gallery JS when needed
We should focus on removing large, rarely-used dependencies from the main course bundle. I.e., all course pages can load the same JS, except video and image-gallery pages should load extra files. (Videojs and its dependencies, along with nanogallery, make up the vast majority of the course JS bundle. It's at least 75%, but I think closer to 90%).

So, IMO:
### Acceptance Criteria
- [x] Only load VideoJS and its dependencies on pages that use VideoJS
- [x] Only load nanogallery2 and its dependencies (there are none) on pages that use it.

### Suggested Approaches
I have two ideas for how to approach this:

1. **At build time:** Put VideoJS in a separate Webpack entrypoint. When building a page, figure out if uses VideoJS. (E.g., if a partial uses VideoJS, have it write to some context like `.Page.Scratch`.). If VideoJS is used, add the appropriate link/script tags.
2. **Dynamic imports:** Webpack is very good at splitting dynamic imports into separate chunks and loading them only as needed. So if the partials call something like `initVideo()`, a function defined in webpack assets whose body includes `import("video.js").then(() => { /* init code */ })`, then `video.js` (and other potential dependencies) would not be loaded unless `initVideo()` is called.

Option 2 seems a little simpler, but Option 1 is nice because it would be determined at built time, and that really seems like it should be the goal.

### PR DESCRIPTION
**Pull request by ibrahimjaved12**
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets? 
#648 

#### What's this PR do?
This PR implements lazy loading for videojs and its dependencies using dynamic imports. It also uses promise-polyfill for IE11 support.

#### How should this be manually tested?
We will have to test working of videojs (including on Internet Explorer 11). You can test it on any course that has lecture videos. Make sure your cache is cleared first.
As an example, you can use [this course.](https://github.mit.edu/ocw-content-rc/9.40-spring-2018) for testing locally.

1. Checkout this branch
2. `yarn start course`
3. Go to "[Lecture Videos](http://localhost:3000/video_galleries/lecture-videos/)"
4. Play any video. Make sure it plays.

**To test for Internet Explorer 11:**
1. Login to browserstack live.
2. Select Internet Explorer 11
3. Use [this netflify link](https://ocw-hugo-themes-course-v2-pr-1089--ocw-next.netlify.app/resources/part-1-introduction-and-overview-of-business-plans/) and play the video. Make sure it plays.

### CODE REVIEW

**Comment by ibrahimjaved12** [comment]
This PR (dynamic import functionality) is on hold because of webpack config need to be updated: https://github.com/mitodl/ocw-hugo-themes/pull/1102
But I believe the work here is done for this PR.
Once the webpack config PR is merged with main, I'll update this PR with instructions to test.

**Review by HussainTaj-arbisoft — CHANGES_REQUESTED** [review]
I ran into a bit of trouble while following the instructions. In the future, it would be great if you can mention the course id that we should test the change on. It would also be great if you could share some instructions about the environment, like "test in incognito" or "clear cache".

File: base-theme/layouts/partials/youtube_player.html
Comment (ibrahimjaved12) [reply base-theme/layouts/partials/youtube_player.html]: Done

**Comment by ibrahimjaved12** [comment]
Thanks for the suggestions @HussainTaj-arbisoft. I have also updated the post with the testing suggestions.

**Review by HussainTaj-arbisoft — APPROVED** [review]
LGTM :+1:

**Pull request merged**
